### PR TITLE
docs: update `root_validator` references to `model_validator` in v2

### PR DESCRIPTION
**Pull request by themattmorris**
## Change Summary

Some documentation references to `root_validator` from `v1` have persisted, so I updated them to `model_validator`.  The changes made in this PR do not impact functionality.

## Related issue number

NA

## Checklist

* [ ] Unit tests for the changes exist
* [X] Tests pass on CI and coverage remains at 100%
* [X] Documentation reflects the changes where applicable
* [X] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @Kludex

### CODE REVIEW

**Comment by themattmorris** [comment]
please review

File: docs/usage/validators.md [hunk 1]
```diff
@@ -450,12 +450,12 @@ except ValidationError as e:
     """
 ```
 
-As with field validators, root validators can have `pre=True`, in which case they're called before field
+As with field validators, model validators can have `pre=True`, in which case they're called before field
```
Comment (ybressler) [inline docs/usage/validators.md:453]: Do model validators use `pre=True`? I thought V2 introduced `mode="before|after"`

File: docs/usage/validators.md [hunk 2]
```diff
@@ -450,12 +450,12 @@ except ValidationError as e:
...
 validation occurs (and are provided with the raw input data), or `pre=False` (the default), in which case
 they're called after field validation.
 
-Field validation will not occur if `pre=True` root validators raise an error. As with field validators,
-"post" (i.e. `pre=False`) root validators by default will be called even if prior validators fail; this
+Field validation will not occur if `pre=True` model validators raise an error. As with field validators,
+"post" (i.e. `pre=False`) model validators by default will be called even if prior validators fail; this
```
Comment (ybressler) [inline docs/usage/validators.md:458]: Same comment here – `pre` becomes `mode`

**Review by ybressler — CHANGES_REQUESTED** [review]
Small clarification required for `pre` vs. `mode`

File: docs/usage/validators.md (hunk 1 above)
Comment (Kludex) [reply docs/usage/validators.md:453]: Nop, they don't use pre anymore.

Comment (themattmorris) [reply docs/usage/validators.md:453]: Thanks for catching this.  I can make the updates, but I don't actually understand how `mode='wrap'` works or if there is an equivalent to `skip_on_failure` anymore.  Can someone keep me honest on these proposed changes, namely if this is what `mode='wrap'` actually does?

```md
As with field validators, model validators can have `mode='before'`, in which case they're called before field
validation occurs (and are provided with the raw input data) or `mode='after'`, in which case
they're called after field validation.

Field validation will not occur if `mode='before'` model validators raise an error. As with field validators,
"post" (i.e. `mode='after'`) model validators by default will be called even if prior validators fail; this
behaviour can be changed by using `mode='wrap'`. The `values` argument will be a dict containing the values which passed field validation and
field defaults where applicable.
```

Comment (ybressler) [reply docs/usage/validators.md:453]: `wrap` appears to be an annotated validator – one which will expect a validator to catch exceptions for validation errors and return a default value instead: https://docs.pydantic.dev/latest/api/functional_validators/#pydantic.functional_validators.WrapValidator

Lemme run a quick test to see the behavior of `wrap`.

Comment (ybressler) [reply docs/usage/validators.md:453]: Okay, so I used this code to test stuff out:
```py3
from typing import Optional

from pydantic import BaseModel, Field, field_validator


def validate_char(cls, v) -> Optional[str]:
    # truncate strings to their first character
    if isinstance(v, str) and len(v) > 1:
        return v[0]

    return v


class Model(BaseModel):
    char_1: str = Field(description="A single character", min_length=1, max_length=1)
    char_2: str = Field(description="A single character", min_length=1, max_length=1)
    char_3: str = Field(description="A single character", min_length=1, max_length=1)


    normalize_char_1 = field_validator('char_1', mode="before")(validate_char)
    # normalize_char_2 = field_validator('char_2', mode="after")(validate_char)
    normalize_char_3 = field_validator('char_2', mode="wrap")(validate_char)


if __name__ == '__main__':
    my_model = Model(
        char_1="AAA",
        char_2="BBB",
        char_3="CCC"
    )

    print(my_model)

```
Raises the following error:
> pydantic.errors.PydanticUserError: Unrecognized field_validator function signature for <bound method validate_char of <class '__main__.Model'>> with `mode=wrap`:(v) -> Optional[str]

> For further information visit https://errors.pydantic.dev/2.0.2/u/validator-signature

So this expects a function which expects a wrapper function.

## Wrapped validator
I rewrote some stuff to match the expected signature:
```py
from typing import Annotated, Optional

from pydantic import BaseModel, Field, ValidationError, WrapValidator, field_validator


def validate_char(cls, v) -> Optional[str]:
    # truncate strings to their first character
    if isinstance(v, str) and len(v) > 1:
        return v[0]

    return v


def wrapped_validate_char(v, handler) -> Optional[str]:
    try:
        return validate_char(None, v)  # Try custom validator
    except ValidationError:
        return handler(v)  # Pydantic validator
    finally:
        return "X"  # Manually set default value

class Model(BaseModel):
    char_1: str = Field(description="A single character", min_length=1, max_length=1)
    char_2: str = Field(description="A single character", min_length=1, max_length=1)

    # Wrap can only be set on the annotation, not on the field validator function
    char_3: Annotated[str, WrapValidator(wrapped_validate_char)] = Field(description="A single character", min_length=1, max_length=1)


if __name__ == '__main__':
    my_model = Model(
        char_1="A",
        char_2="B",
        char_3="CCC"
    )

    print(my_model)
```

Comment (ybressler) [reply docs/usage/validators.md:453]: TLDR: Mode `wraps` should not be possible on the decorator. (Unless I'm wrong.) It lets you explicitly wrap Pydantic's validator function.

**Review by tpdorsey — APPROVED** [review]

File: changes/6471-themattmorris.md
Comment (Kludex) [inline changes/6471-themattmorris.md:1]: We can remove this file.

**Review by Kludex — CHANGES_REQUESTED** [review]

**Review by adriangb — APPROVED** [review]

**Review by adriangb — APPROVED** [review]

**Review by adriangb — APPROVED** [review]

**Pull request merged**